File: Infinity.Bnois.Web/Infinity.Bnois.Web/app/services/dashboardService.js
(function () {
    'use strict';
    angular.module('app').service('dashboardService', ['dataConstants', 'apiHttpService', dashboardService]);

    function dashboardService(dataConstants, apiHttpService) {
        var service = {
            getDashboardOutSideNavy: getDashboardOutSideNavy,
            getDashboardUnMission: getDashboardUnMission,
            getDashboardBranch: getDashboardBranch,
            getDashboardAdminAuthority: getDashboardAdminAuthority,
            getDashboardInsideNavyOrganization: getDashboardInsideNavyOrganization,
            getDashboardBCGOrganization: getDashboardBCGOrganization,
            getDashboardLeave: getDashboardLeave,
            getDashboardExBDLeave: getDashboardExBDLeave,
            getDashboardStream: getDashboardStream,
            getDashboardCategory: getDashboardCategory,
            getDashboardGender: getDashboardGender,
            getDashboardOfficeAppointment: getDashboardOfficeAppointment,
            getDashboardUnderCourse: getDashboardUnderCourse,
            getDashboardUnderMission: getDashboardUnderMission,
            getOutsideNavyOfficer: getOutsideNavyOfficer,
            getAdminAuthorityOfficer: getAdminAuthorityOfficer,
            getLeaveOfficer: getLeaveOfficer,
            getExBDLeaveOfficer: getExBDLeaveOfficer,
            getBranchOfficer: getBranchOfficer,
            getStreamOfficer: getStreamOfficer,
            getCategoryOfficer: getCategoryOfficer,
            getOfficersByOffice: getOfficersByOffice,
            getOfficerByAdminAuthorityWithDynamicQuery: getOfficerByAdminAuthorityWithDynamicQuery,
            getBranchAuthorityOfficers9: getBranchAuthorityOfficers9,
            getBranchAuthorityOfficers10: getBranchAuthorityOfficers10,
            getBranchAuthorityOfficers11: getBranchAuthorityOfficers11,
            getBranchAuthorityOfficers12: getBranchAuthorityOfficers12,
            getBranchAuthorityOfficers13: getBranchAuthorityOfficers13,
            getBranchAuthorityOfficers369: getBranchAuthorityOfficers369,
            getBranchAuthorityOfficers383: getBranchAuthorityOfficers383,
            getBranchAuthorityOfficers458: getBranchAuthorityOfficers458,
            getBranchAuthorityOfficers513: getBranchAuthorityOfficers513,
            getBranchAuthorityOfficers543: getBranchAuthorityOfficers543,
            getBranchAuthorityOfficers600: getBranchAuthorityOfficers600,
            getBNOfficerStates950: getBNOfficerStates950,
            getToeOfficerStateInNavy: getToeOfficerStateInNavy,
            getToeOfficerStateInside: getToeOfficerStateInside,
            getGenderOfficer: getGenderOfficer,
            getGenderOfficerByAuthority: getGenderOfficerByAuthority,
            getToeOfficerListByTransferType: getToeOfficerListByTransferType,
            GetOverviewOfficerDeploymentList: GetOverviewOfficerDeploymentList
           
           
        };

        return service;
     
        function getDashboardOutSideNavy(officeId) {
            var url = dataConstants.DASHBOARD_URL + 'get-dashboard-outside-navy?officeId=' + officeId;
            return apiHttpService.GET(url);
        }


        function getDashboardUnMission(officeId) {
            var url = dataConstants.DASHBOARD_URL + 'get-dashboard-un-mission?officeId=' + officeId;
            return apiHttpService.GET(url);
        }

        function getDashboardAdminAuthority() {
            var url = dataConstants.DASHBOARD_URL + 'get-dashboard-admin-authority';
            return apiHttpService.GET(url);
        }
        function getDashboardInsideNavyOrganization() {
            var url = dataConstants.DASHBOARD_URL + 'get-dashboard-inside-navy-organization';
            return apiHttpService.GET(url);
        }
         function getDashboardBCGOrganization() {
            var url = dataConstants.DASHBOARD_URL + 'get-dashboard-bcg-organization';
            return apiHttpService.GET(url);
        }

       
        function getDashboardBranch() {
            var url = dataConstants.DASHBOARD_URL + 'get-dashboard-branch';
            return apiHttpService.GET(url);
        }

        function getDashboardLeave() {
            var url = dataConstants.DASHBOARD_URL + 'get-dashboard-leave';
            return apiHttpService.GET(url);
        }

       function getDashboardExBDLeave() {
            var url = dataConstants.DASHBOARD_URL + 'get-dashboard-ex-bd-leave';
            return apiHttpService.GET(url);
        }

       
        function getDashboardStream(streamId) {
            var url = dataConstants.DASHBOARD_URL + 'get-dashboard-stream?streamId='+streamId;
            return apiHttpService.GET(url);
        }


        function getDashboardCategory(categoryId) {
            var url = dataConstants.DASHBOARD_URL + 'get-dashboard-category?categoryId=' + categoryId;
            return apiHttpService.GET(url);
        }

        function getDashboardGender(genderId) {
            var url = dataConstants.DASHBOARD_URL + 'get-dashboard-gender?genderId=' + genderId;
            return apiHttpService.GET(url);
        }


        function getDashboardOfficeAppointment(officeType, rankId, displayId) {
            var url = dataConstants.DASHBOARD_URL + 'get-dashboard-office-appointment?officeType=' + officeType + '&rankId=' + rankId + '&displayId=' + displayId;
            return apiHttpService.GET(url);
        }

        function getDashboardUnderMission(rankId) {
            var url = dataConstants.DASHBOARD_URL + 'get-dashboard-under-mission?rankId=' + rankId;
            return apiHttpService.GET(url);
        }

        function getDashboardUnderCourse(rankId) {
            var url = dataConstants.DASHBOARD_URL + 'get-dashboard-under-course?rankId=' + rankId;
            return apiHttpService.GET(url);
        }



        function getOutsideNavyOfficer(officeId, rankLevel, parentId) {
            var url = dataConstants.DASHBOARD_URL + 'get-outside-navy-officer?officeId=' + officeId + '&rankLevel=' + rankLevel + '&parentId=' + parentId;
            return apiHttpService.GET(url);
        }
        function getOfficersByOffice(officeId) {
            var url = dataConstants.DASHBOARD_URL + 'get-office-search-result?officeId=' + officeId;
            return apiHttpService.GET(url);
        }

        function getAdminAuthorityOfficer(officeId, rankLevel,type) {
            var url = dataConstants.DASHBOARD_URL + 'get-admin-authority-officer?officeId=' + officeId+'&rankLevel='+rankLevel+'&type='+type;
            return apiHttpService.GET(url);
        }


        function getLeaveOfficer(leaveType, rankLevel) {
            var url = dataConstants.DASHBOARD_URL + 'get-leave-officer?leaveType=' + leaveType+'&rankLevel='+rankLevel;
            return apiHttpService.GET(url);
        }

        function getExBDLeaveOfficer( rankLevel) {
            var url = dataConstants.DASHBOARD_URL + 'get-ex-bd-leave-officer?rankLevel='+rankLevel;
            return apiHttpService.GET(url);
        }


        function getBranchOfficer(rankId, branch, categoryId, subCategoryId, commissionTypeId) {
            var url = dataConstants.DASHBOARD_URL + 'get-branch-officer?rankId=' + rankId + '&branch=' + branch + '&categoryId=' + categoryId + '&subCategoryId=' + subCategoryId + '&commissionTypeId=' + commissionTypeId;
            return apiHttpService.GET(url);
        }


        function getStreamOfficer(rankId, branch, streamId) {
            var url = dataConstants.DASHBOARD_URL + 'get-stream-officer?rankId=' + rankId + '&branch=' + branch + '&streamId=' + streamId;
            return apiHttpService.GET(url);
        }

        function getCategoryOfficer(rankId, branch, categoryId) {
            var url = dataConstants.DASHBOARD_URL + 'get-category-officer?rankId=' + rankId + '&branch=' + branch + '&categoryId=' + categoryId;
            return apiHttpService.GET(url);
        }

        function getGenderOfficer(rankId, branch, categoryId, subCategoryId, commissionTypeId, genderId) {
            var url = dataConstants.DASHBOARD_URL + 'get-gender-officer?rankId=' + rankId + '&branch=' + branch + '&categoryId=' + categoryId + '&subCategoryId=' + subCategoryId + '&commissionTypeId=' + commissionTypeId + '&genderId=' + genderId;
            return apiHttpService.GET(url);
        }
        function getGenderOfficerByAuthority(adminAuthorityId,rankId, branch, categoryId, subCategoryId, commissionTypeId, genderId) {
            var url = dataConstants.DASHBOARD_URL + 'get-branch-officer-by-admin-authority?adminAuthorityId=' + adminAuthorityId +'&rankId=' + rankId + '&branch=' + branch + '&categoryId=' + categoryId + '&subCategoryId=' + subCategoryId + '&commissionTypeId=' + commissionTypeId + '&genderId=' + genderId;
            return apiHttpService.GET(url);
        }
        function getToeOfficerListByTransferType(rankId, branch, categoryId, subCategoryId, commissionTypeId, transferType) {
            var url = dataConstants.DASHBOARD_URL + 'toe-officer-by-transfer-type?rankId=' + rankId + '&branch=' + branch + '&categoryId=' + categoryId + '&subCategoryId=' + subCategoryId + '&commissionTypeId=' + commissionTypeId + '&transferType=' + transferType;
            return apiHttpService.GET(url);
        }
        function GetOverviewOfficerDeploymentList(rankId, officerTypeId, coastGuard, outsideOrg) {
            var url = dataConstants.DASHBOARD_URL + 'get-overview-officer-deployment-list?rankId=' + rankId + '&officerTypeId=' + officerTypeId + '&coastGuard=' + coastGuard + '&outsideOrg=' + outsideOrg;
            return apiHttpService.GET(url);
        }
        function getOfficerByAdminAuthorityWithDynamicQuery(tableName,branchAuthorityId) {
            var url = dataConstants.DASHBOARD_URL + 'get-officer-by-admin-authority-with-dynamic-query?tableName=' + tableName + '&branchAuthorityId=' + branchAuthorityId;
            return apiHttpService.GET(url);
        }
        function getBranchAuthorityOfficers9(branchAuthorityId) {
            var url = dataConstants.DASHBOARD_URL + 'get-branch-authority-officer-9?branchAuthorityId=' + branchAuthorityId;
            return apiHttpService.GET(url);
        }
        function getBranchAuthorityOfficers10(branchAuthorityId) {
            var url = dataConstants.DASHBOARD_URL + 'get-branch-authority-officer-10?branchAuthorityId=' + branchAuthorityId;
            return apiHttpService.GET(url);
        }
        function getBranchAuthorityOfficers11(branchAuthorityId) {
            var url = dataConstants.DASHBOARD_URL + 'get-branch-authority-officer-11?branchAuthorityId=' + branchAuthorityId;
            return apiHttpService.GET(url);
        }
        function getBranchAuthorityOfficers12(branchAuthorityId) {
            var url = dataConstants.DASHBOARD_URL + 'get-branch-authority-officer-12?branchAuthorityId=' + branchAuthorityId;
            return apiHttpService.GET(url);
        }
        function getBranchAuthorityOfficers13(branchAuthorityId) {
            var url = dataConstants.DASHBOARD_URL + 'get-branch-authority-officer-13?branchAuthorityId=' + branchAuthorityId;
            return apiHttpService.GET(url);
        }
        function getBranchAuthorityOfficers369(branchAuthorityId) {
            var url = dataConstants.DASHBOARD_URL + 'get-branch-authority-officer-369?branchAuthorityId=' + branchAuthorityId;
            return apiHttpService.GET(url);
        }
        function getBranchAuthorityOfficers383(branchAuthorityId) {
            var url = dataConstants.DASHBOARD_URL + 'get-branch-authority-officer-383?branchAuthorityId=' + branchAuthorityId;
            return apiHttpService.GET(url);
        }
        function getBranchAuthorityOfficers458(branchAuthorityId) {
            var url = dataConstants.DASHBOARD_URL + 'get-branch-authority-officer-458?branchAuthorityId=' + branchAuthorityId;
            return apiHttpService.GET(url);
        }
        function getBranchAuthorityOfficers513(branchAuthorityId) {
            var url = dataConstants.DASHBOARD_URL + 'get-branch-authority-officer-513?branchAuthorityId=' + branchAuthorityId;
            return apiHttpService.GET(url);
        }
        function getBranchAuthorityOfficers543(branchAuthorityId) {
            var url = dataConstants.DASHBOARD_URL + 'get-branch-authority-officer-543?branchAuthorityId=' + branchAuthorityId;
            return apiHttpService.GET(url);
        }
        function getBranchAuthorityOfficers600() {
            var url = dataConstants.DASHBOARD_URL + 'get-branch-authority-officer-600';
            return apiHttpService.GET(url);
        }
        function getBNOfficerStates950() {
            var url = dataConstants.DASHBOARD_URL + 'get-bn-officer-states-950';
            return apiHttpService.GET(url);
        }
        function getToeOfficerStateInNavy() {
            var url = dataConstants.DASHBOARD_URL + 'get-toe-officer-state-in-navy';
            return apiHttpService.GET(url);
        }
        function getToeOfficerStateInside() {
            var url = dataConstants.DASHBOARD_URL + 'get-toe-officer-state-inside';
            return apiHttpService.GET(url);
        }
       

    }
})();

File: Infinity.Bnois/Infinity.Bnois.ApplicationService/Implementation/DashboardService.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.IO;
using System.Linq;
using System.Threading.Tasks;
using Infinity.Bnois.ApplicationService.Interface;
using Infinity.Bnois.Data;

namespace Infinity.Bnois.ApplicationService.Implementation
{
    public class DashboardService : IDashboardService
    {

        private readonly IBnoisRepository<Employee> employeeRepository;

        public DashboardService(IBnoisRepository<Employee> employeeRepository)
        {
            this.employeeRepository = employeeRepository;

        }



        public List<object> GetDashboardOutSideNavy(int officeId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashboardOutSideNavy] {0} ", officeId));

            return dataTable.ToJson().ToList();
        }

        public List<object> GetDashboardUnMission(int officeId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardUnMission] {0} ", officeId));

            return dataTable.ToJson().ToList();
        }

        public List<object> GetDashboardBranch()
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashboardBranch]"));

            return dataTable.ToJson().ToList();
        }
        public List<object> GetDashboardAdminAuthority()
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashboardAdminAuthority]"));

            return dataTable.ToJson().ToList();
        }
        public List<object> GetDashboardUnderMission(int rankId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashboardUnderMission] {0}", rankId));

            return dataTable.ToJson().ToList();
        }
        public List<object> GetDashboardUnderCourse(int rankId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashboardUnderCourse] {0}", rankId));

            return dataTable.ToJson().ToList();
        }
        public List<object> GetDashboardInsideNavyOrganization()
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardInsideNavyOrganization]"));

            return dataTable.ToJson().ToList();

        }
        public List<object> GetDashboardBCGOrganization()
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardBCGOrganization]"));

            return dataTable.ToJson().ToList();
        }
        public List<object> GetDashboardLeave()
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashboardLeave]"));

            return dataTable.ToJson().ToList();
        }


        public List<object> GetDashboardExBDLeave()
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashboardExBDLeave]"));

            return dataTable.ToJson().ToList();
        }

        public List<object> GetDashboardStream(int streamId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashboardStream] {0}",streamId));

            return dataTable.ToJson().ToList();
        }

        public List<object> GetDashboardCategory(int categoryId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashboardCategory] {0}",categoryId));

            return dataTable.ToJson().ToList();
        }

        public List<object> GetDashboardGender(int genderId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashboardGender] {0}", genderId));

            return dataTable.ToJson().ToList();
        }
        

        public List<object> GetBranchAuthorityOfficer9(int branchAuthorityId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardBranchByAdminAuthority9] {0}", branchAuthorityId));

            return dataTable.ToJson().ToList();
        }
        public List<object> GetBranchAuthorityOfficer10(int branchAuthorityId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardBranchByAdminAuthority10] {0}", branchAuthorityId));

            return dataTable.ToJson().ToList();
        }
        public List<object> GetBranchAuthorityOfficer11(int branchAuthorityId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardBranchByAdminAuthority11] {0}", branchAuthorityId));

            return dataTable.ToJson().ToList();
        }
        public List<object> GetBranchAuthorityOfficer12(int branchAuthorityId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardBranchByAdminAuthority12] {0}", branchAuthorityId));

            return dataTable.ToJson().ToList();
        }
        public List<object> GetBranchAuthorityOfficer13(int branchAuthorityId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardBranchByAdminAuthority13] {0}", branchAuthorityId));

            return dataTable.ToJson().ToList();
        }
        public List<object> GetBranchAuthorityOfficer369(int branchAuthorityId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardBranchByAdminAuthority369] {0}", branchAuthorityId));

            return dataTable.ToJson().ToList();
        }
        public List<object> GetBranchAuthorityOfficer383(int branchAuthorityId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardBranchByAdminAuthority383] {0}", branchAuthorityId));

            return dataTable.ToJson().ToList();
        }
        
        public List<object> GetBranchAuthorityOfficer458(int branchAuthorityId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardBranchByAdminAuthority458] {0}", branchAuthorityId));

            return dataTable.ToJson().ToList();
        }
        public List<object> GetBranchAuthorityOfficer513(int branchAuthorityId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardBranchByAdminAuthority513] {0}", branchAuthorityId));

            return dataTable.ToJson().ToList();
        }
        public List<object> GetBranchAuthorityOfficer543(int branchAuthorityId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardBranchByAdminAuthority543] {0}", branchAuthorityId));

            return dataTable.ToJson().ToList();
        }
        public List<object> GetOfficerByAdminAuthorityWithDynamicQuery(string tableName,int branchAuthorityId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetOfficerByAdminAuthorityWithDynamicQuery] '{0}',{1}", tableName, branchAuthorityId));

            return dataTable.ToJson().ToList();
        }
        public List<object> GetBranchAuthorityOfficer600()
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardBranchByAdminAuthority600]"));

            return dataTable.ToJson().ToList();
        }
        public List<object> getBNOfficerStates950()
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardBranch950]"));

            return dataTable.ToJson().ToList();
        }
        public List<object> getToeOfficerStateInside()
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardBranch800]"));

            return dataTable.ToJson().ToList();
        }
        public List<object> getToeOfficerStateInNavy()
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashBoardBranch900]"));

            return dataTable.ToJson().ToList();
        }

        public List<object> GetDashboardOfficeAppointment(int officeType, int rankId, int displayId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDashboardOfficeAppointment] {0},{1},{2}", officeType,rankId,displayId));

            return dataTable.ToJson().ToList();
        }

        public List<object> GetOutsideNavyOfficer(int officeId, int rankLevel,int parentId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetOutsideNavyOfficer] {0},{1},{2}", officeId,rankLevel, parentId));

            return dataTable.ToJson().ToList();
        }
        

        public List<object> GetOfficeSearchResult(int officeId)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetEmployeeListByOffice] {0}", officeId));

            return dataTable.ToJson().ToList();
        }

        public List<object> GetAdminAuthorityOfficer(int officeId, int rankLevel,int type)
        {
            if (type == 1)
            {
                DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetAdminAuthorityOfficer] {0},{1}", officeId, rankLevel));
                return dataTable.ToJson().ToList();
            }
            else if (type == 2)
            {
                DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetInsideNavyOfficer] {0},{1}", officeId, rankLevel));
                return dataTable.ToJson().ToList();
            }
            else
            {
                DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetBCGOfficer] {0},{1}", officeId, rankLevel));
                return dataTable.ToJson().ToList();
            }
           
        }

        public List<object> GetLeaveOfficer(int leaveType, int rankLevel)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetLeaveOfficer] {0},{1}", leaveType, rankLevel));

            return dataTable.ToJson().ToList();
        }

        public List<object> GetExBDLeaveOfficer(int rankLevel)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetExBDLeaveOfficer] {0}", rankLevel));

            return dataTable.ToJson().ToList();
        }

        public List<object> GetBranchOfficer(int rankId, string branch,int categoryId, int subCategoryId, int commissionTypeId)
        {
            string query= String.Format("exec [spGetBranchOfficer] {0},'{1}',{2},{3},{4}", rankId, branch, categoryId, subCategoryId, commissionTypeId);
            if (branch == null)
            {
                query = String.Format("exec [spGetBranchOfficer] {0},null, {1},{2},{3}", rankId, categoryId, subCategoryId, commissionTypeId);
            }

            DataTable dataTable = employeeRepository.ExecWithSqlQuery(query);
            
            return dataTable.ToJson().ToList();
        }

        public List<object> GetBranchOfficerByAdminAuthority(int addAuthId, int rankId, string branch,int categoryId, int subCategoryId, int commissionTypeId)
        {
            string query= String.Format("exec [spGetBranchOfficerByAdminAuthority] {0},{1},'{2}',{3},{4},{5}", addAuthId, rankId, branch, categoryId, subCategoryId, commissionTypeId);
            if (branch == null)
            {
                query = String.Format("exec [spGetBranchOfficerByAdminAuthority] {0}, {1},null,{2},{3},{4}", addAuthId, rankId, categoryId, subCategoryId, commissionTypeId);
            }

            DataTable dataTable = employeeRepository.ExecWithSqlQuery(query);
            
            return dataTable.ToJson().ToList();
        }
        public List<object> GetStreamOfficer(int rankId, string branch,int streamId)
        {
            string query = String.Format("exec [spGetStreamOfficer] {0},'{1}',{2}", rankId, branch, streamId);

            if (branch == null)
            {
                query = String.Format("exec [spGetStreamOfficer] {0},null,{1}", rankId, streamId);
            }
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(query);

            return dataTable.ToJson().ToList();
        }
        public List<object> GetOverviewOfficerDeploymentList(int rankId, int officerTypeId, int coastGuard, int outsideOrg)
        {
            string query = String.Format("exec [spGetOverviewOfOfficerDeploymentList] {0},{1},{2},{3}", rankId, officerTypeId, coastGuard, outsideOrg);

            if (outsideOrg == 100)
            {
                if (coastGuard == 100)
                {
                    if (officerTypeId == 100)
                    {
                        query = String.Format("exec [spGetOverviewOfOfficerDeploymentList] {0},null,null,null", rankId);
                    }
                    else
                    {
                        query = String.Format("exec [spGetOverviewOfOfficerDeploymentList] {0},{1},null,null", rankId, officerTypeId);
                    }
                }
                else
                {
                    query = String.Format("exec [spGetOverviewOfOfficerDeploymentList] {0},{1},{2},null", rankId, officerTypeId, coastGuard);
                } 
            }
            //else
            //{
            //    query = String.Format("exec [spGetOverviewOfOfficerDeploymentList] {0},{1},{2},{3}", rankId, officerTypeId, coastGuard, outsideOrg);
            //}
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(query);

            return dataTable.ToJson().ToList();
        }

        public List<object> GetCategoryOfficer(int rankId, string branch, int categoryId)
        {
            string query = String.Format("exec [spGetCategoryOfficer] {0},'{1}',{2}", rankId, branch, categoryId);

            if (branch == null)
            {
                query = String.Format("exec [spGetCategoryOfficer] {0},null,{1}", rankId, categoryId);
            }
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(query);

            return dataTable.ToJson().ToList();
        }

        public List<object> GetGenderOfficer(int rankId, string branch,int categoryId,int subCategoryId,int commissionTypeId, int genderId)
        {
            string query = String.Format("exec [spGetGenderOfficer] {0},'{1}',{2},{3},{4},{5}", rankId, branch, categoryId, subCategoryId, commissionTypeId, genderId);

            if (branch == null)
            {
                query = String.Format("exec [spGetGenderOfficer] {0},null,{1},{2},{3},{4}", rankId, categoryId, subCategoryId, commissionTypeId, genderId);
            }
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(query);

            return dataTable.ToJson().ToList();
        }

        public List<object> GetToeOfficerByTransferType(int rankId, string branch,int categoryId,int subCategoryId,int commissionTypeId, int transferType)
        {
            string query = String.Format("exec [spGetToeOfficerByTransferType] {0},'{1}',{2},{3},{4},{5}", rankId, branch, categoryId, subCategoryId, commissionTypeId, transferType);

            if (branch == null)
            {
                query = String.Format("exec [spGetToeOfficerByTransferType] {0},null,{1},{2},{3},{4}", rankId, categoryId, subCategoryId, commissionTypeId, transferType);
            }
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(query);

            return dataTable.ToJson().ToList();
        }
    }
}